lib/validates_zipcode/formatter.rb

```diff
@@ -0,0 +1,50 @@
+module ValidatesZipcode
+  module Formatter
+    class << self
+      # we assume the zipcode is valid for that country
+      def format(zipcode, country_alpha2)
+        method = "format_#{country_alpha2.to_s.downcase}"
+        respond_to?(method) ? send(method, zipcode) : format_default(zipcode)
```

Comment:
I don't really like using `respond_to?`, it can lead to unintended consequences. 

At least, I'd prefer to whitelist the alpha_2. Maybe by raising an `ArgumentError` in the initializer `unless ZIPCODES_REGEX.keys.include?(@country_alpha2)`

Reply:
Still, I'd prefer to use a hash with lambas. Something like this: 

```ruby 
ZIPCODES_TRANSFORMATIONS = {
  DE: ->(z) { z.rjust(5, '0') },
  AT: ->(z) { z.scan(/d/).join },
  ....  
}.tap { |hash| hash.default = Proc.new(&:strip) }.freeze
```

and then 

```ruby 
def format
  ZIPCODES_TRANSFORMATIONS[@country_alpha2].call(@zipcode)
end
```

Reply:
OK, I see your point about aliases. You could do something like this: 

```ruby
ZIPCODES_TRANSFORMATIONS = {
  DE: ->(z) { z.rjust(5, '0') },
  AT: ->(z) { z.scan(/d/).join },
  UK: method(:format_uk),
  GB: method(:format_uk)
  ....  
}.tap { |hash| hash.default = Proc.new(&:strip) }.freeze

def format_uk(zipcode)
  zipcode.upcase.scan(/[A-Z0-9]/).insert(-4, ' ').join
end
```

A bit hackish, maybe.

Maybe a reason to get back to methods format.

Reply:
Ok, I think I have found a way with a case-statement. It is now possible to give a proc or a Symbol that references to another proc in the transformations hash. What do you think?

Reply:
Elegant solution 👍